### DISCUSSION TITLE
Approximate PR/ROC AUC Metics On Large Data During Training

### DISCUSSION THREAD

**Discussion by EricZimmermann** [Project-MONAI/MONAI#2982] [Ideas]
Hello, 

During my work, I've encountered issues when aiming to compute PR AUC on large MRI volumes during training without impacting train time. To account for this, I've written some code to approximate these curves quickly on the gpu as a Ignite Metric. I assume I can estimate the upper and lower bounds of my models output and use these estimates to discretize the integral over predefined thresholds. 

Is this code worth contributing?

**Comment by wyli** [comment]
would be nice to have a pull request for discussion.  reminds me of this code from @yiheng-wang-nv  https://github.com/Project-MONAI/tutorials/blob/77382cdb36fd9b99a693e2588a769fbc4f69cb06/kaggle/RANZCR/4th_place_solution/utils.py#L104-L116

**Reply by EricZimmermann** [reply]
It is somewhat similar although our method although ours would aggregate stats over the entire training set using an accumulator. Should I open the PR?

**Reply by wyli** [reply]
If it's a generic PR AUC module, another option is to contribute directly to Ignite https://github.com/pytorch/ignite/tree/master/ignite/metrics

(cc @vfdev-5)

**Reply by vfdev-5** [reply]
@wyli thanks for pinging ! @EricZimmermann yes, I agree if it is a generic metric, Ignite would be happy to make it available to all our users as well. Feel free to send a PR to https://github.com/pytorch/ignite. Thanks!

**Reply by EricZimmermann** [reply]
Sorry for the delayed response, I've been quite busy and making a PR will take some time. 
I wanted to highlight that what I'd be contributing would be an **approximate** metric that really only shines for very large data (mostly brains) and would be suited for people wanting to remove memory bottlenecks when doing segmentation and optimizing their model across a variety of operating points. Would it be possible to post or send a code snippet before going through the whole PR process? Thank you so much for your time!

**Reply by vfdev-5** [reply]
@EricZimmermann sure, no worries about the time. Yes, we can start llke that or maybe just send a draft PR to iterate over later.

**Reply by EricZimmermann** [reply]
I can make a public colab and share the link if that would be okay

**Reply by vfdev-5** [reply]
Sounds good :+1:

**Reply by EricZimmermann** [reply]
Here it is! 
Let me know what you think! It seems to have been useful in our lab thus far.

https://colab.research.google.com/drive/17Xo80swTxcaZsZOKS08GI-LHYHkg5kud?usp=sharing
